lib/logging/index.js

```diff
@@ -0,0 +1,543 @@
+
+// jscs:disable maximumLineLength
+/**
+ * Create a sink.
+ *
+ * **This method only works if you are authenticated as yourself, e.g. using the
+ * gcloud SDK.**
+ *
+ * @resource [Sink Overview]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.sinks}
+ * @resource [Advanced Logs Filters]{@link https://cloud.google.com/logging/docs/view/advanced_filters}
+ * @resource [projects.sinks.create API Documentation]{@link https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.sinks/create}
+ *
+ * @throws {Error} if a name is not provided.
+ * @throws {Error} if a config object is not provided.
+ *
+ * @param {string} name - Name of the sink.
+ * @param {object} config - See a
+ *     [Sink resource](https://cloud.google.com/logging/docs/api/ref/rest/v1beta3/projects.logServices.sinks#LogSink).
+ * @param {module:storage/bucket|module:bigquery/dataset|module:pubsub/topic} config.destination -
+ *     The destination. If a GCS Bucket object is provided, the proper ACL scope
+ *     will be granted to the bucket.
+ * @param {string=} config.filter - An advanced logs filter. Only log entries
+ *     matching the filter are written.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:loggin/sink} callback.sink - The created Sink object.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * var gcs = gcloud.storage();
+ *
+ * var config = {
+ *   destination: gcs.bucket('logging-bucket'),
+ *   filter: 'metadata.severity = ALERT'
+ * };
+ *
+ * function callback(err, sink, apiResponse) {
+ *   // `sink` is a Sink object.
+ * }
+ *
+ * logging.createSink('new-sink-name', config, callback);
+ */
+Logging.prototype.createSink = function(name, config, callback) {
+  // jscs:enable maximumLineLength
+  var self = this;
+
+  if (!is.string(name)) {
+    throw new Error('A sink name must be provided.');
+  }
+
+  if (!is.object(config)) {
+    throw new Error('A sink configuration object must be provided.');
+  }
+
+  if (config.destination instanceof Bucket) {
+    var bucket = config.destination;
+
+    // Set the proper permissions on the bucket.
+    bucket.acl.owners.addGroup('cloud-logs@google.com', function(err, apiResp) {
+      if (err) {
+        callback(err, null, apiResp);
+        return;
+      }
+
+      config.destination = 'storage.googleapis.com/' + bucket.name;
+
+      self.createSink(name, config, callback);
+    });
+
+    return;
+  }
+
+  if (config.destination instanceof Dataset) {
+    var dataset = config.destination;
+
+    config.destination = format('{baseUrl}/projects/{pId}/datasets/{dId}', {
+      baseUrl: 'bigquery.googleapis.com',
+      pId: dataset.parent.projectId,
+      dId: dataset.id
+    });
+  }
+
+  if (config.destination instanceof Topic) {
+    var topic = config.destination;
+
+    config.destination = format('{baseUrl}/{topicName}', {
```

Comment:
Yep, `topic.name` is its formatted name: https://github.com/GoogleCloudPlatform/gcloud-node/blob/1ac48c3177010f04c72acfcf7f36c76f45cadacc/lib/pubsub/topic.js#L216